rj_gameplay/rj_gameplay/play/penalty_defense.py

```diff
@@ -1,137 +1,68 @@
+    """
+    Play consists of:
+                    -5 robots lined up on the side
+                    -1 goalie
+    The goalie will defend the opponents penalty kick.
+    """
 
     def __init__(self):
-        self.tactics = [
-            goalie_tactic.GoalieTactic(),
-            move_tactic.Move((1.6, 9.0)),
-            move_tactic.Move((1.9, 9.0)),
-            move_tactic.Move((2.2, 9.0)),
-            move_tactic.Move((2.5, 9.0)),
-            move_tactic.Move((2.8, 9.0)),
-        ]
-
-        self.role_assigner = NaiveRoleAssignment()
-
-    def compute_props(self, prev_props):
-        pass
+        super().__init__()
+        self._state = State.INIT
 
     def tick(
         self,
-        world_state: rc.WorldState,
-        prev_results: role.assignment.FlatRoleResults,
-        props,
-    ) -> Tuple[
-        Dict[Type[tactic.SkillEntry], List[role.RoleRequest]],
-        List[tactic.SkillEntry],
-    ]:
-
-        # Get role requests from all tactics and put them into a dictionary
-        role_requests: play.RoleRequests = {
-            tactic: tactic.get_requests(world_state, None) for tactic in self.tactics
-        }
-
-        # Flatten requests and use role assigner on them
-        flat_requests = play.flatten_requests(role_requests)
-        flat_results = self.role_assigner.assign_roles(
-            flat_requests, world_state, prev_results
-        )
-        role_results = play.unflatten_results(flat_results)
-
-        # Get list of all SkillEntries from all tactics
-        skills = []
-        for tactic in self.tactics:
-            skills += tactic.tick(world_state, role_results[tactic])
-
-        # Get all role assignments
-        # SkillEntry to (list of?) RoleResult
-        skill_dict = {}
-        for tactic in self.tactics:
-            skill_dict.update(role_results[tactic])
-
-        return (skill_dict, skills)
+        world_state: stp.rc.WorldState,
+    ) -> List[RobotIntent]:
+        if self._state == State.INIT:
+            self.prioritized_tactics.append(goalie_tactic.GoalieTactic(world_state, 0))
+            self.prioritized_tactics.append(line_tactic.LineTactic(world_state, 5))
+            self.assign_roles(world_state)
+            self._state = State.ACTIVE
+            return self.get_robot_intents(world_state)
+        elif self._state == State.ACTIVE:
+            return self.get_robot_intents(world_state)
+
+
+class PreparePenaltyDefense(stp.play.Play):
+    def __init__(self):
+        super().__init__()
+        self._state = State.INIT
 
-    def is_done(self, world_state):
-        # returns done when all tactics are done
-        # TODO: change all is_done() to use all()?
-        return all([tactic.is_done(world_state) for tactic in self.tactics])
+    def tick(
+        self,
+        world_state: stp.rc.WorldState,
+    ) -> List[RobotIntent]:
+        if self._state == State.INIT:
+            self.prioritized_tactics.append(line_tactic.LineTactic(world_state, 5))
```

Comment:
I believe we can start assigning the goalie at this stage since the goalie has nothing to do with the "1m rule".  Correct me if I'm wrong

Reply:
That makes sense. I will also add goalieTactic to prioritized tactics.